### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by RaphDasilva**
## Changes
- Created a new branch (Interactive list)
- Add linter for HMTL, CSS, and JavaScript 
- Set up a new project with webpack
- Added a new JavaScript file and import it as a module
-Added event listener to the checkbox.
- Updated items object's value for completed key upon user actions.
- Implemented a function for the "Clear all completed" button (used filter() method).
- Stored the updated array of items in local storage, so the user gets the correct list values after the page reloads.
- Created a function for refreshing the page
- Rendered the dynamically created list of tasks in the dedicated placeholder.
- Add a README file 
- Made use of ES6 variable declaration formats in the projects:
- Made use of ES6 arrow functions.
- Fixed linter errors

## Checklist before requesting a review
- [X] Liner errors 
- [X] GitHub flow
- [X] Documented professionally 
- [X] Followed HTML & CSS best practices 
- [X] Followed JavaScript best practices

### CODE REVIEW

File: src/modules/check-box.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const checkBox = (box) => {
+  if (box.checked) {
+    return true;
+  }
+  return false;
+};
+
+export default checkBox;
```
Comment (thecodechaser) [inline src/modules/check-box.js:8]: - [ ] When I make tasks as completed(By ticking the checkboxes) and refresh the page, I don't see the checkboxes ticked even I the local storage values are correct to true. Please make sure to fix this issue and refreshing the page should show tasks as completed(Ticked checkboxes) for tasks marked as completed before the page resfresh.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @RaphDasilva👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/check-box.js (hunk 1 above)
Comment (RaphDasilva) [reply src/modules/check-box.js:8]: @thecodechaser But this is optional....is not on the requirement.

Comment (Nemwel-Boniface) [inline src/modules/check-box.js:8]: - [ ] As requested of you by the previous reviewer "_When I make tasks as completed(By ticking the checkboxes) and refresh the page, I don't see the checkboxes ticked even I the local storage values are correct to true. Please make sure to fix this issue and refreshing the page should show tasks as completed(Ticked checkboxes) for tasks marked as completed before the page refresh_" Kindly implement this to meet all the project requirements.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @RaphDasilva         :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/check-box.js (hunk 1 above)
Comment (Nemwel-Boniface) [reply src/modules/check-box.js:8]: Hello @RaphDasilva  this is part of the requirements. 

image

When a task is marked as completed, it is good to make sure that even if the user leaves now and comes back later and reloads, he/ she is supposed to still be able to see that a completed task is marked differently from other tasks.

Comment (RaphDasilva) [reply src/modules/check-box.js:8]: @Nemwel-Boniface  But this is optional....is not on the requirement.

Comment (RaphDasilva) [reply src/modules/check-box.js:8]: Hello....

Comment (RaphDasilva) [reply src/modules/check-box.js:8]: This is yours.... https://nemwel-boniface.github.io/todoList

Comment (RaphDasilva) [reply src/modules/check-box.js:8]: Check it and see

**Comment by RaphDasilva** [comment]
Hello

**Comment by medaminedev66** [comment]
Hello @RaphDasilva 👋,

_This is an additional review as a result of the second opinion request._

It's one of the requirements that the user should get the correct list values after the page reloads:

image

Additionally, for a better experience, the user should be aware of which tasks have already been completed so they do not need to be checked again. For example, if the value of "completed" is true in the local storage but appears to be unchecked on the page, when the user checks it, it will be false in the local storage but checked on the page, which is incongruous.

I recommend implementing the changes requested by the previous reviewers so that your project is complete.

Keep rocking 🚀!

**Review by BregornOriginal — APPROVED** [review]
## Hi @RaphDasilva 

## Your project is complete!  There is nothing else to say other than... it's time to merge it 

## Congratulations! 

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Excelent work making changes requested by the previous reviewer 💯 💯 💯 

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**